docs/executable-plugins.md

```diff
@@ -0,0 +1,88 @@
+---
+title: Executable Plugins
+description: Plugins built as stand-alone applications can be written in languages such as Python, and may be more familiar to some developers.
+---
+
+MyST is able to invoke plugins written in different languages through standard IO protocols. Executable MyST plugins are treated as a [](wiki:Black_box), whereby MyST only sees the data it passes to the plugin, and the response from the plugin itself.
+
+
+## Defining a new directive
+
+:::{note}
+There are many different ways to create an executable plugin. Here we'll use Python to implement an `unsplash-py` directive, but any programming language that can read and write from stdin, stdout, and stderr, and define a commandline interface would work.
+:::
+
+First, we'll declare the plugin _specification_ that allows MyST to discover the directives, transforms, and/or roles that the plugin implements. This specification looks very similar to [the definition of a JavaScript plugin](javascript-plugins.md#unsplash-js-source), except the implementation logic (e.g. the directive `run` method) is not defined.
+
+:::{literalinclude} unsplash.py
+:caption: a plugin to add an `unsplash-py` directive that includes a beautiful, random picture based on a query string.
+:::
+this code should be referenced from your `myst.yml` under the `projects.plugins`:
+
+```{code} yaml
+:filename: myst.yml
+project:
+  plugins:
+    - type: executable
+      path: unsplash.mjs
```

Comment:
Should this be py?

Reply:
Haha yes, I noticed that the other day. About to board a flight - could you tweak?